receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper.go

```diff
@@ -111,26 +114,40 @@ func initializeNetworkDataPoint(dataPoint pdata.Int64DataPoint, startTime pdata.
 	dataPoint.SetValue(value)
 }
 
-func scrapeAndAppendNetworkTCPConnectionsMetric(metrics pdata.MetricSlice) error {
-	connections, err := net.Connections("tcp")
+func (s *scraper) scrapeAndAppendNetworkTCPConnectionsMetric(metrics pdata.MetricSlice) error {
+	connections, err := s.connections("tcp")
 	if err != nil {
 		return err
 	}
 
-	connectionStatusCounts := getConnectionStatusCounts(connections)
+	connectionStatusCounts := getTCPConnectionStatusCounts(connections)
 
 	startIdx := metrics.Len()
 	metrics.Resize(startIdx + 1)
 	initializeNetworkTCPConnectionsMetric(metrics.At(startIdx), connectionStatusCounts)
 	return nil
 }
 
-func getConnectionStatusCounts(connections []net.ConnectionStat) map[string]int64 {
-	connectionStatuses := make(map[string]int64, len(connections))
+func getTCPConnectionStatusCounts(connections []net.ConnectionStat) map[string]int64 {
+	var tcpStatuses = map[string]int64{
```

Comment:
Why not an array and a enum for status? Seems better for me

Reply:
I'm not totally sure what you mean by this.

Do you mean enum as in?

```go
type tcpStatus string

const (
	closeWait   tcpStatus = "CLOSE_WAIT"
	closed      tcpStatus = "CLOSED"
	...
)
```

Also I think a map should outperform an array here since we need to iterate through potentially 100s of connections to record the counts. Using an array would require O(n^2)?

Reply:
You may need a switch for the status. Not 100% sure so your call

Reply:
Oh okay yea - ofc 🤦‍♂️

We would need a switch to convert strings to indexes, so I suspect performance would be pretty similar either way. Annoyingly gopsutil actually converts these from ints to strings before returning the values 😞 

If you don't feel too strongly about this I will probably leave it as is. This has the nice propery of if we ever receive an unexpected status value, it will get added to the map.